python/interpret_community/explanation/explanation.py

```diff
@@ -513,9 +513,12 @@ def _local_data(self, parent_data, key=None):
         # Note: we currently don't have access to instances on explanation
         parent_data[InterpretData.VALUES] = None
         if key is not None:
+            # Note: eval_y_predicted and eval_data can be id, in which case we should skip this logic
+            has_eval_y_predicted = not isinstance(self._eval_y_predicted, str)
+            has_eval_data = not isinstance(self._eval_data, str)
             # Note: the first argument should be the true y's but we don't have that
             # available currently, using predicted instead for now
-            if _DatasetsMixin._does_quack(self):
+            if _DatasetsMixin._does_quack(self) and has_eval_y_predicted and has_eval_data:
```

Comment:
Should this check be part of `_does_quack`? Does the check for str type apply to all `_DatasetsMixin`?

Reply:
does quack just checks those properties exist, not what the values are.  This change is in the data method.

Reply:
I think it might be good to isolate the logic checking the type of these two properties. These two properties are always related to a `_DatasetsMixin` right? Maybe `_DatasetsMixin` can have another static method that checks whether the dataset is a reference or in-memory data? I imagine something like `_DatasetsMixin.is_reference()`

Reply:
Then in plain English you would say "only update parent_data if this adheres to the `_DatasetsMixin` interface AND it's not a reference type `_DatasetsMixin`"

Reply:
done! added _is_reference() method